### ISSUE TITLE
'hypergan' is not recognized as an internal or external command

### ISSUE THREAD

**Issue by ghost** [HyperGAN/HyperGAN#269]
I installed all of hypergan's requirements and did `pip3 install hypergan` in cmd. No errors. I even tried adding C:\Users\\[user]\AppData\Local\Programs\Python\Python37\Lib\site-packages\hypergan to PATH. Everytime I use hypergan in command prompt, I get the error `'hypergan' is not recognized as an internal or external command,
operable program or batch file.`

Screenshot of error:
[https://i.imgur.com/bQP3aGy.png](url)

OS: Windows 10

Any help is greatly appreciated.

**Comment by martyn** [comment]
Hi, it looks like the program is not in the path. I'm not sure why this is (we develop on linux) but windows support is top of the list after multi gpu support. I will tag this/similar tickets with 'windows support'. Our windows installer will resolve this.

**Comment by IIIBlueberry** [comment]
it was probably due to removal of hypergan.cmd that exist on version 0.10
https://github.com/HyperGAN/HyperGAN/tree/0.10.1/bin

**Comment by martyn** [comment]
Good catch! This was a mistake so I've re-added the hypergan.cmd and re-released a new pip package.

**Comment by owcschafer** [comment]
Hi, I've just installed HyperGAN today on Windows and I'm getting the same error. I used `pip3 install hypergan` and it installed the latest version with no issues. However, running `hypergan train preset:celeba -s 128x128x3` to test it out gives the same error. Is there something that needs to be fixed or am I doing something wrong? Thanks!

**Comment by axel578** [comment]
Same for me doesn't work

**Comment by joajoajpedroj** [comment]
Running `pip3 install hypergan` on PowerShell as administrator seems to have solved this. However, when I run `hypergan` this "how do you wish to open this file" screen pops up
![image](https://user-images.githubusercontent.com/69732445/94561280-2bbb8380-023a-11eb-9fbe-1075b8857dbb.png)

**Comment by badcode6** [comment]
Same problem, renaming it to hypergan.py and running it with py hypergan.py makes it say hypergan is not a package

**Comment by Konradoslaw** [comment]
You need to add manually the "hypergan.cmd" file, that @IIIBlueberry mentioned, to the anaconda3\envs\[YOUR ENVIRONMENT]\Scripts folder (on Win10 and Anaconda).

For some reason, pip still does not add this file during installation both on Windows 10 and MacOS (at least on my friend's MacBook).

**Comment by MasterCubo** [comment]
> Hi, I've just installed HyperGAN today on Windows and I'm getting the same error. I used `pip3 install hypergan` and it installed the latest version with no issues. However, running `hypergan train preset:celeba -s 128x128x3` to test it out gives the same error. Is there something that needs to be fixed or am I doing something wrong? Thanks!

same exact problem. Just installed with pip. @Konradoslaw where do I find said cmd file?

**Comment by MasterCubo** [comment]
> You need to add manually the "hypergan.cmd" file, that @IIIBlueberry mentioned, to the anaconda3\envs\YOUR ENVIRONMENT\Scripts folder (on Win10 and Anaconda).
> 
> For some reason, pip still does not add this file during installation both on Windows 10 and MacOS (at least on my friend's MacBook).

also, that directory isn't familiar to me, I do not use anaconda. What is the parallel here?

**Comment by Hedgy134117** [comment]
I've got the same exact problem as described here. I tried directly running the hypergan file in the directory, same error.

**Comment by MasterCubo** [comment]
> I've got the same exact problem as described here. I tried directly running the hypergan file in the directory, same error.

Hello! If you join the discord, I had a really helpful person send me their files and work me through it and I'm pretty sure the main problem was with dependencies. Try doing `pip3 freeze` and sending me the results.

**Comment by BigSweaterMan** [comment]
> You need to add manually the "hypergan.cmd" file, that @ IIIBlueberry mentioned, to the anaconda3\envs\YOUR ENVIRONMENT\Scripts folder (on Win10 and Anaconda).
> 
> For some reason, pip still does not add this file during installation both on Windows 10 and MacOS (at least on my friend's MacBook).

I have the same problem, how do I add hypergan.cmd? I am almost completely lost on how to solve this issue.

**Comment by ghost** [comment]
bump
still not working a year later

**Comment by lovebug0706** [comment]
bump, still not working.

**Comment by martyn** [comment]
The repeated reports after the old `.cmd` re-release make clear that the launcher problem was not reliably resolved. The new `develop` foundation now uses a standard Python console-script entrypoint and includes `python -m hypergan`; #301 tests installed wheels on Windows, macOS and Linux, including the Windows launcher. The Windows-specific test failure found during this work was fixed and rerun successfully.

Closing this against the implemented development foundation. It has **not** been published as a new PyPI release, so `pip install hypergan` can still retrieve the historical package discussed here. To try the new lightweight commands, follow the [development checkout installation](https://github.com/HyperGAN/HyperGAN/blob/develop/README.md), activate its environment, run `python -m pip install .`, then `python -m hypergan --help`. Do not add `site-packages/hypergan` to PATH or copy launcher files by hand. This establishes lightweight Windows commands, not Windows GPU/image training support.

Audit record: [individual disposition and retained work](https://github.com/HyperGAN/HyperGAN/blob/f4053ae1/reports/issue-audit-2026-09-18.md), published in #302.
